tests/devices/qutrit_mixed/test_qutrit_mixed_apply_operation.py

```diff
@@ -0,0 +1,307 @@
+        """A dummy debugger class"""
+
+        def __init__(self):
+            self.active = True
+            self.snapshots = {}
+
+    @pytest.mark.usefixtures("two_qutrit_state")
+    def test_no_debugger(
+        self, ml_framework, state, shape, request
+    ):  # pylint: disable=unused-argument
+        """Test nothing happens when there is no debugger"""
+        state = request.getfixturevalue(state)
+        initial_state = math.asarray(state, like=ml_framework)
+
+        new_state = apply_operation(qml.Snapshot(), initial_state, is_state_batched=len(shape) != 2)
+        assert new_state.shape == initial_state.shape
+        assert math.allclose(new_state, initial_state)
+
+    def test_empty_tag(self, ml_framework, state, shape, request):
+        """Test a snapshot is recorded properly when there is no tag"""
+        state = request.getfixturevalue(state)
+        initial_state = math.asarray(state, like=ml_framework)
+
+        debugger = self.Debugger()
+        new_state = apply_operation(
+            qml.Snapshot(), initial_state, debugger=debugger, is_state_batched=len(shape) != 2
+        )
+
+        assert new_state.shape == initial_state.shape
+        assert math.allclose(new_state, initial_state)
+
+        assert list(debugger.snapshots.keys()) == [0]
+        assert debugger.snapshots[0].shape == shape
+        assert math.allclose(debugger.snapshots[0], math.reshape(initial_state, shape))
+
+    def test_provided_tag(self, ml_framework, state, shape, request):
+        """Test a snapshot is recorded property when provided a tag"""
+        state = request.getfixturevalue(state)
+        initial_state = math.asarray(state, like=ml_framework)
+
+        debugger = self.Debugger()
+        tag = "dense"
+        new_state = apply_operation(
+            qml.Snapshot(tag), initial_state, debugger=debugger, is_state_batched=len(shape) != 2
+        )
+
+        assert new_state.shape == initial_state.shape
+        assert math.allclose(new_state, initial_state)
+
+        assert list(debugger.snapshots.keys()) == [tag]
+        assert debugger.snapshots[tag].shape == shape
+        assert math.allclose(debugger.snapshots[tag], math.reshape(initial_state, shape))
+
+    def test_snapshot_with_measurement(self, ml_framework, state, shape, request):
+        """Test a snapshot with measurement throws NotImplementedError"""
+        state = request.getfixturevalue(state)
+        initial_state = math.asarray(state, like=ml_framework)
+
+        debugger = self.Debugger()
+        with pytest.raises(NotImplementedError):
+            _ = apply_operation(
+                qml.Snapshot(measurement=qml.expval(qml.GellMann(0, 1))),
+                initial_state,
+                debugger=debugger,
+                is_state_batched=len(shape) != 2,
+            )
+
+
+@pytest.mark.parametrize("ml_framework", ml_frameworks_list)
+class TestBroadcasting:  # pylint: disable=too-few-public-methods
```

Comment:
```suggestion
class TestOperation:  # pylint: disable=too-few-public-methods
```

Can rename to reflect contents; it looks like this class tests all possible combinations of batched/unbatched operations/states? On that note, the unbroadcasted op and unbroadcasted state (most basic) case does not seem to be covered in this class or elsewhere (unless I missed it?).

Reply:
It was however it was removed, good point I will update it to be included here.